src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -717,10 +717,10 @@ static void window_footpath_set_provisional_path_at_point(int32_t x, int32_t y)
     TileElement* tileElement{};
     LocationXY16 mapCoord = {};
     get_map_coordinates_from_pos(
-        x, y, VIEWPORT_INTERACTION_MASK_FOOTPATH & VIEWPORT_INTERACTION_MASK_TERRAIN, &mapCoord.x, &mapCoord.y,
-        &interactionType, &tileElement, nullptr);
-    x = mapCoord.x;
-    y = mapCoord.y;
+        screenCoords.x, screenCoords.y, VIEWPORT_INTERACTION_MASK_FOOTPATH & VIEWPORT_INTERACTION_MASK_TERRAIN, &mapCoord.x,
+        &mapCoord.y, &interactionType, &tileElement, nullptr);
+    screenCoords.x = mapCoord.x;
```

Comment:
No don't do this. Screen cords and map cords should never mix.

Reply:
What do you suggest? I mean because the assignment was already there, it just became more explicit.

Reply:
Is the assignment actually needed? Can't you just use mapcoord directly. The aim of this refactoring is to clearly indicate what is screen and what is map.

Reply:
I don't think it is strictly needed, I'll update it and we'll see.